src/core/ext/transport/chttp2/transport/http2_server_transport.h

```diff
@@ -0,0 +1,67 @@
+//
+//
+// Copyright 2024 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#ifndef GRPC_SRC_CORE_EXT_TRANSPORT_CHTTP2_TRANSPORT_HTTP2_SERVER_TRANSPORT_H
+#define GRPC_SRC_CORE_EXT_TRANSPORT_CHTTP2_TRANSPORT_HTTP2_SERVER_TRANSPORT_H
+
+#include "src/core/ext/transport/chttp2/transport/hpack_encoder.h"
+#include "src/core/ext/transport/chttp2/transport/hpack_parser.h"
+#include "src/core/lib/transport/promise_endpoint.h"
+#include "src/core/lib/transport/transport.h"
+
+namespace grpc_core {
+namespace http2 {
+
+// Experimental : This is just the initial skeleton of class
+// and it is functions. The code will be written iteratively.
+// Do not use or edit any of these functions unless you are
+// familiar with the PH2 project (Moving chttp2 to promises.)
+// TODO(tjagtap) : [PH2][P3] : Delete this comment when http2
+// rollout begins
+class Http2ServerTransport final : public ServerTransport {
+ public:
+  Http2ServerTransport(
+      GRPC_UNUSED PromiseEndpoint control_endpoint,
+      GRPC_UNUSED PromiseEndpoint data_endpoint,
+      GRPC_UNUSED const ChannelArgs& args,
+      GRPC_UNUSED std::shared_ptr<grpc_event_engine::experimental::EventEngine>
+          event_engine,
+      GRPC_UNUSED HPackParser hpack_parser,
```

Comment:
HPackParser and HPackCompressor coming in as parameters is something ChaoticGood needs (it exchanges frames before transport creation), but not something that our http2 stack needs, so we can omit these arguments.

Reply:
Done